src/main/java/org/codehaus/mojo/flatten/FlattenMode.java

```diff
@@ -72,7 +72,11 @@
     clean,
 
     /** Removes all {@link FlattenDescriptor optional POM elements} and dependencies. */
-    fatjar;
+    fatjar,
+    
+    /** Only resolves variables revision, sha and changelist. Keeps everything else. */
+    resolveCiFriendliesOnly;

```

Comment:
I have no knowledge about the origins of the "Maven CI Friendly Versions" documentation, so I cannot answer about it. The purpose is to just resolve the variables that are stated there, but I think mostly the use case would be for "revision".
"sha" should be "sha1".
I am fine with any naming for the variable.

Reply:
Just to make sure there is no misunderstanding (I am awaiting this PR): the guys at the head of Maven (@khmarbaise & al) themselves came up with 3 harcoded variable names that are possible to be accepted as variable part of the version. This is what is usually called "CI Friendly Maven". This may not be the best choice, there may have been different alternative implementations, but to this day this is the current status of Core Maven. Knowing the forward compatible history of Maven, I doubt they will change. So there is really no need to try to find alternatives here. Even non core plugin should just reuse what Maven user are (or will be) familiar with, the way of naming things (CIFriendly, sha1, changelist & revision) should be the same.
My 2 cents

Reply:
@fmarot could you shed in some light here? Do you agree to some of my points or do you have some arguments for your PoV?

Reply:
@lasselindqvist as you are so superb fast and active in updating the PR:
would it be possible that you link to the original documentation from the javadoc then like this:

See <a href="https://maven.apache.org/maven-ci-friendly.html">Maven CI Friendly</a> for further details.

Reply:
yes, each subject one after another:

- resolveCiFriendliesOnly vs resolveCiVariablesOnly : I prefer the former to keep it linked with the Maven Core vocabulary of "CI Friendly". But it is not very important
- sha1 VS sha in the comment : @lasselindqvist already corrected it
- rename sha1 to a more generic word: this is a no go as Maven Core uses explicitly "sha1" as one of the 3 allowed keywords.
- changelist : same as above. the Flatten plugin has no choice but to comply with the wording of Maven Core. Otherwise users would not understand.
(just trying to help here ;) thanks for your work both of you)

Reply:
@fmarot thanks for the quick feedback. I was not aware if this is just something a single guy came up with - but if maven core already adopted this I do not want to fight anything here. I just want to care about design and quality and avoide foolish decisions ;)
@khmarbaise can you confirm this.
IMHO we could then merge this and maybe I might even be able to push out a new release before xmas.

Reply:
@mfriedenhagen confirmed on dev list:
https://groups.google.com/forum/#!topic/mojohaus-dev/1SQSAERVg1c